### PR TITLE
feat(service-portal): hms loans

### PR DESCRIPTION
**Pull request by Valur**
## What

Display loans from HMS in service portal

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/hms-loans/src/lib/dto/getHmsLoansPaymenthistory.input.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Field, InputType } from '@nestjs/graphql'
+import { IsNumber } from 'class-validator'
+
+@InputType()
+export class GetHmsLoansPaymenthistoryInput {
```
Comment (thorkellmani) [inline libs/api/domains/hms-loans/src/lib/dto/getHmsLoansPaymenthistory.input.ts:5]:
```suggestion
export class GetHmsLoansPaymentHistoryInput {
```

the h is annoyingly lower case

File: libs/api/domains/hms-loans/src/lib/models/loanHistoryPdf.model.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Field, ObjectType } from '@nestjs/graphql'
+
+@ObjectType('HmsLoansLoanhistoryPdf')
+export class LoanhistoryPdf {
```
Comment (thorkellmani) [inline libs/api/domains/hms-loans/src/lib/models/loanHistoryPdf.model.ts:4]: lower case h 😮‍💨

File: libs/api/domains/hms-loans/src/lib/models/loanHistory.model.ts [hunk 3]
```diff
@@ -27,12 +27,14 @@ export class CoPayer {
   coPayerNationalId?: string
 }
 
-@ObjectType('HmsLoansLoan')
+@ObjectType('HmsLoansLoanhistory')
```
Comment (thorkellmani) [inline libs/api/domains/hms-loans/src/lib/models/loanHistory.model.ts:30]: lower case h 😮‍💨

File: libs/api/domains/hms-loans/src/lib/models/paymenthistory.model.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { Field, ObjectType } from '@nestjs/graphql'
 
-@ObjectType('HmsLoansPayment')
+@ObjectType('HmsLoansPaymenthistory')
```
Comment (thorkellmani) [inline libs/api/domains/hms-loans/src/lib/models/paymenthistory.model.ts:3]: ow, lower case h 😮‍💨

File: libs/service-portal/finance/src/components/FinanceLoans/FinanceLoansTable.tsx [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+  ] = useGetHmsLoansLoanhistoryPdfLazyQuery({
+    onCompleted: (data) => {
+      if (data.hmsLoansLoanhistoryPdf?.data) {
+        const downloadLink = document.createElement('a')
+        downloadLink.href = `data:${
+          data.hmsLoansLoanhistoryPdf.mime ?? 'application/pdf'
+        };base64,${data.hmsLoansLoanhistoryPdf.data}`
+        downloadLink.download =
+          data.hmsLoansLoanhistoryPdf.name ?? 'lanayfirlit.pdf'
+        downloadLink.click()
```
Comment (thorkellmani) [inline libs/service-portal/finance/src/components/FinanceLoans/FinanceLoansTable.tsx:44]: might be worth it to extract this into a hook or util

File: libs/service-portal/finance/src/lib/messages.ts [hunk 6]
```diff
@@ -13,4 +13,65 @@ export const m = defineMessages({
...
+    defaultMessage: 'Kostnaður',
+  },
+  priceImprovementPayment: {
+    id: 'sp.finance-loans:cost-payment',
+    defaultMessage: 'Verð afborgun',
+  },
+  priceImprovementInterest: {
+    id: 'sp.finance-loans:cost-payment',
+    defaultMessage: 'Verð vextir',
+  },
```
Comment (thorkellmani) [inline libs/service-portal/finance/src/lib/messages.ts:62]: Repeated translation string ids

File: libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.graphql [hunk 7]
```diff
@@ -0,0 +1,81 @@
+query GetHmsLoansLoanhistory {
+  hmsLoansLoanhistory {
```
Comment (thorkellmani) [inline libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.graphql:2]: lowercase h owie 🤕

File: libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.graphql [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    coPayers {
+      coPayerName
+      coPayerNationalId
+    }
+  }
+}
+
+query GetHmsLoansLoanhistoryPdf {
+  hmsLoansLoanhistoryPdf {
```
Comment (thorkellmani) [inline libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.graphql:61]: lowercase h owie 🤕

File: libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.graphql [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    mime
+    data
+    name
+  }
+}
+
+query GetHmsLoansPaymenthistory($input: GetHmsLoansPaymenthistoryInput!) {
+  hmsLoansPaymenthistory(input: $input) {
```
Comment (thorkellmani) [inline libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.graphql:69]: lowercase h owie 🤕

File: libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.graphql (hunk 7 above)
Comment (thorkellmani) [reply libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.graphql:2]: Also, you don't have to repeat the same word in singular/plural while naming the queries. `HmsLoansHistory` should cover it. If it doesn't, might we worth a rethink on the modelling, i think

File: libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.tsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+const FinanceLoans = () => {
+  useNamespaces('sp.finance-loans')
+  const { formatMessage } = useLocale()
+
+  const {
+    data: loanOverviewData,
+    loading: loanOverviewLoading,
+    error: loanOverviewError,
+    called: loanOverviewCalled,
+  } = useGetHmsLoansLoanhistoryQuery()
```
Comment (thorkellmani) [inline libs/service-portal/finance/src/screens/FinanceLoans/FinanceLoans.tsx:18]: Ow, lowercase h 🤕

**Review by thorkellmani — CHANGES_REQUESTED** [review]
Mostly all good, except the very annoying lowercase `history`

File: libs/clients/hms-loans/src/lib/hmsLoans.provider.ts [hunk 11]
```diff
@@ -33,6 +33,7 @@ export const HmsLoansApiProvider: Provider<LibraApi> = {
                 scope: [HmsScope.loans],
               }
             : undefined,
+          timeout: 60000,
```
Comment (thorkellmani) [inline libs/clients/hms-loans/src/lib/hmsLoans.provider.ts:36]: wow, why the extreme timeout?

File: libs/service-portal/finance/src/lib/messages.ts [hunk 12]
```diff
@@ -13,4 +13,65 @@ export const m = defineMessages({
...
+    id: 'sp.finance-loans:transaction-date',
+    defaultMessage: 'Hreyfingadagur',
+  },
+  firstPaymentDate: {
+    id: 'sp.finance-loans:first-payment-date',
+    defaultMessage: 'Fyrsti gjalddagi',
+  },
+  originalAmount: {
+    id: 'sp.finance-loans:first-payment-date',
+    defaultMessage: 'Upphafleg fjárhæð',
```
Comment (thorkellmani) [inline libs/service-portal/finance/src/lib/messages.ts:44]: Same id

**Review by thorkellmani — CHANGES_REQUESTED** [review]
Just one question and one repeated message id

Also, try to avoid repeating the same word singular/plural in the grahpql object names, fx HmsLoansLoanHistory -> HmsLoansHistory should be enough

File: libs/clients/hms-loans/src/lib/hmsLoans.provider.ts (hunk 11 above)
Comment (Valur) [reply libs/clients/hms-loans/src/lib/hmsLoans.provider.ts]: ugh, the service is very slow on dev/staging. They say it's much faster on prod.

**Review by thorkellmani — APPROVED** [review]
LGTM

**Pull request merged**